packages/core/aqua-compiler/src/generate/index.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CompilationResult, JSTypeGenerator, OutputType, TSTypeGenerator } from './interfaces.js';
+import { getPackageJsonContent } from '../utils.js';
+import { ServiceGenerator } from './service.js';
+import { FunctionGenerator } from './function.js';
+import header from './header.js';
+
+export async function generateSources({ services, functions }: CompilationResult, outputType: OutputType) {
+    const typeGenerator = outputType === 'js' ? new JSTypeGenerator() : new TSTypeGenerator();
+    const { version, devDependencies } = await getPackageJsonContent();
```

Comment:
Why not put this directly into the header function as I previously suggested? It will remove the need for duplication. You have the same exact code in `generateTypes` function

Reply:
I don't like nesting async function too deep

Reply:
please check again